code/game/objects/structures.dm

```diff
@@ -16,9 +16,7 @@
 		cameranet.updateVisibility(src)
 
 /obj/structure/blob_act(obj/effect/blob/B)
-	if(!density)
-		qdel(src)
-	if(prob(50))
+	if(density && prob(50))
 		qdel(src)

```

Comment:
Actually totally intended, nondense stuff survives in the blob for blah blah something something reasons.